modules/json/src/smithy4s/json/internals/SchemaVisitorJCodec.scala

```diff
@@ -1299,35 +1336,81 @@ private[smithy4s] class SchemaVisitorJCodec(
       )
   }
 
+  trait UnknownFieldsEncoder[A] {
```

Comment:
Rather than creating a bespoke trait, I'd delegate to the `Document.Encoder` to produce a document representation. I'd then match the document to check if it's a DObject and would call upon a function to inline its fields, that I'd factor out of the encoder implementation for documents.

Reply:
Done in https://github.com/disneystreaming/smithy4s/commit/59e0d225da28823f1679bc8b7243e0deab05a75a. Wasn't sure what you meant by this:
> I'd factor out of the encoder implementation for documents
I think you're referring to this:
```
          case Document.DObject(values) =>
            values.foreach { case (key, value) =>
              out.writeKey(key)
              documentJCodec.encodeValue(value, out)
            }
```
but it seems too small to be worth factoring out to me - can you confirm?